test/js/css-url-image-value.js

```diff
@@ -0,0 +1,47 @@
+suite('CSSURLImageValue', function() {
+  test('CSSURLImageValue is a CSSURLImageValue, CSSImageValue, CSSResourceValue, and CSSStyleValue', function() {
+    assert.instanceOf(new CSSURLImageValue('resources/1x1-green.png'), CSSURLImageValue);
+    assert.instanceOf(new CSSURLImageValue('resources/1x1-green.png'), CSSImageValue);
+    assert.instanceOf(new CSSURLImageValue('resources/1x1-green.png'), CSSResourceValue);
+    assert.instanceOf(new CSSURLImageValue('resources/1x1-green.png'), CSSStyleValue);
+  });
+
+  test('CSSURLImageValue only accepts string', function() {
+    assert.throw(function() { new CSSURLImageValue(); }, TypeError, "URL must be a string");
+    assert.throw(function() { new CSSURLImageValue([]); }, TypeError, "URL must be a string");
+    assert.throw(function() { new CSSURLImageValue(1); }, TypeError, "URL must be a string");
+    assert.doesNotThrow(function() { new CSSURLImageValue(''); });
+  });
+
+  test('URL, state, and intrinsic dimensions of CSSURLImageValue object are correct', function(done) {
+    var urlImageValue = new CSSURLImageValue('resources/1x1-green.png');
+    assert.equal(urlImageValue.url, "resources/1x1-green.png");
+
+    var oldOnload = urlImageValue._image.onload;
+
+    urlImageValue._image.onload = function() {
+      oldOnload();
+      assert.strictEqual(urlImageValue.state, "loaded");
+      assert.strictEqual(urlImageValue.intrinsicWidth, 1);
+      assert.strictEqual(urlImageValue.intrinsicHeight, 1);
+      assert.strictEqual(urlImageValue.intrinsicRatio, 1);
+      done();
+    }
+  });
+
+  test('Invalid image will have error state and null intrinsic dimensions', function(done) {
+    var urlImageValue = new CSSURLImageValue('http://localhost');
```

Comment:
Just use something you know definitely won't return an image. (e.g. resources/nonexisting.png?) Who knows what localhost could return on someone else's machine!